www/docs/interests/category.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+// Logic page for the register of interests by category view
+// if no selected category, will just show a list of the categories
+
+include_once '../../includes/easyparliament/init.php';
+
+$chamber = get_http_var("chamber", "house-of-commons");
+$date = get_http_var("date", null);
+$selected_category_id = get_http_var("category_id", null);
+
+// ensure $date if set is a valid iso date
+if ($date && !preg_match('/^\d{4}-\d{2}-\d{2}$/', $date)) {
+    $date = null;
+}
+
+if ($chamber == 'house-of-commons') {
+    $this_page = "interest_category";
+} elseif ($chamber == 'scottish-parliament') {
+    $this_page = "interest_category_sp";
+} elseif ($chamber == 'senedd') {
+    $this_page = "interest_category_wp";
+} elseif ($chamber == 'northern-ireland-assembly') {
+    $this_page = "interest_category_ni";
+} else {
+    $this_page = "interest_category";
+}
+
+// load a register
+if ($date) {
+    $register = MySociety\TheyWorkForYou\DataClass\Regmem\Register::latestAsOfDate($chamber, $date);
+} else {
+    $register = MySociety\TheyWorkForYou\DataClass\Regmem\Register::getLatest($chamber);
+}
+
+
+// get the relevant categories from that register
+// and if there is a selected_category_id reduce the register here
+$categories = [];
+
+foreach ($register->persons as $person) {
```

Comment:
I understand that we have to do it this way because of the way the data is structured but it seems quite inefficient to do this instead of including the list of categories in the data. And does it not also mean that if no one has declared an entry in a category for the current register we'll not show the category?

Reply:
Yeah, this is consequence of choosing a tiered data format (where all the useful data sits under the person and can be sliced for our database). 

Although nothing stops us adding something to the data format (an extra set of category objects under the register) if we thought that was worth sorting out at this point?

Reply:
I don't think it's worth delaying anything at the moment. I do wonder if it would make it a bit more helpful for anyone re-using the data but again, we could presumably add this in later.